waspc/src/Wasp/Generator/ServerGenerator/Auth/OAuthAuthG.hs

```diff
@@ -38,31 +38,32 @@ import Wasp.Generator.ServerGenerator.JsImport (extImportToImportJson)
 import Wasp.Util ((<++>))
 import qualified Wasp.Util as Util
 
-genOAuthAuth :: AS.Auth.Auth -> Generator [FileDraft]
-genOAuthAuth auth
+genOAuthAuth :: AS.AppSpec -> AS.Auth.Auth -> Generator [FileDraft]
+genOAuthAuth spec auth
   | AS.Auth.isExternalAuthEnabled auth =
-      genOAuthHelpers auth
+      genOAuthHelpers spec auth
         <++> genOAuthProvider googleAuthProvider (AS.Auth.google . AS.Auth.methods $ auth)
         <++> genOAuthProvider gitHubAuthProvider (AS.Auth.gitHub . AS.Auth.methods $ auth)
   | otherwise = return []
 
-genOAuthHelpers :: AS.Auth.Auth -> Generator [FileDraft]
-genOAuthHelpers auth =
+genOAuthHelpers :: AS.AppSpec -> AS.Auth.Auth -> Generator [FileDraft]
+genOAuthHelpers spec auth =
   sequence
-    [ genCreateRouter auth,
+    [ genCreateRouter spec auth,
       genTypes auth,
-      return $ C.mkSrcTmplFd [relfile|auth/providers/oauth/init.ts|],
-      return $ C.mkSrcTmplFd [relfile|auth/providers/oauth/defaults.ts|]
+      genDefaults spec,
+      return $ C.mkSrcTmplFd [relfile|auth/providers/oauth/init.ts|]
     ]
 
-genCreateRouter :: AS.Auth.Auth -> Generator FileDraft
-genCreateRouter auth = return $ C.mkTmplFdWithData [relfile|src/auth/providers/oauth/createRouter.ts|] (Just tmplData)
+genCreateRouter :: AS.AppSpec -> AS.Auth.Auth -> Generator FileDraft
+genCreateRouter spec auth = return $ C.mkTmplFdWithData [relfile|src/auth/providers/oauth/createRouter.ts|] (Just tmplData)
   where
     tmplData =
       object
         [ "userEntityUpper" .= (userEntityName :: String),
           "userEntityLower" .= (Util.toLowerFirst userEntityName :: String),
-          "externalAuthEntityLower" .= (Util.toLowerFirst externalAuthEntityName :: String)
+          "externalAuthEntityLower" .= (Util.toLowerFirst externalAuthEntityName :: String),
+          "isPasswordOnUserEntity" .= (doesUserEntityContainField spec "password" == Just True)
```

Comment:
Nit: I would maybe add a local variable `isPasswordOnUserEntity` and use that to make the `tmplData` less busy

Reply:
Updated 👍🏻